packages/replay/src/util/handleRecordingEmit.ts

```diff
@@ -0,0 +1,86 @@
+import { logger } from '@sentry/utils';
+
+import type { ReplayContainer } from '../replay';
+import { saveSession } from '../session/saveSession';
+import type { RecordingEvent } from '../types';
+import { addEvent } from './addEvent';
+
+type RecordingEmitCallback = (event: RecordingEvent, isCheckout?: boolean) => void;
+
+/**
+ * Handler for recording events.
+ *
+ * Adds to event buffer, and has varying flushing behaviors if the event was a checkout.
+ */
+export function getHandleRecordingEmit(replay: ReplayContainer): RecordingEmitCallback {
```

Comment:
optional:
Now that this function is exctracted, should we add a unit test or two for it? wdyt?

Reply:
good point, makes sense to do that! Will think about this on Monday 👍

Reply:
added some tests for this as well!